components/shared_server_code/src/shared/initialization/database.py

```diff
@@ -45,6 +45,7 @@ def create_indexes(database: Database) -> None:
     """Create any indexes."""
     database.datamodels.create_index("timestamp")
     database.reports.create_index("timestamp")
+    database.users.create_index("username", unique=True)
```

Comment:
Do we really need an index? Are we ever going to have so many users that the index is faster than a full collection scan?

Reply:
good point

Reply:
On closer inspection, this is a way to ensure that the username is unique. Since we find users by username and especially now with multiple LDAP servers there is as far as I know no other location where uniqueness of usernames is guaranteed, we might run into problems otherwise.

Reply:
Of course to do it nicely, we should implement a proper workaround (or error message) for the case we get multiple users with identical usernames, but given the propability of this happening, I feel that just crashing is a good temporary solution.
In any case better than in a later stage potentially providing user data of a different.